aequilibrae/paths/traffic_assignment.py

```diff
@@ -179,16 +198,20 @@ def set_algorithm(self, algorithm: str):
         """
 
         # First we instantiate the arrays we will be using over and over
-        if algorithm not in self.all_algorithms:
+        if algorithm.lower() not in self.all_algorithms:
             raise AttributeError(f"Assignment algorithm not available. Choose from: {','.join(self.all_algorithms)}")
 
         if algorithm.lower() == "all-or-nothing":
             self.assignment = allOrNothing(self)
+        elif algorithm.lower() == 'fw':
+            self.assignment = LinearApproximation(self, "frank-wolfe")
         elif algorithm.lower() in ["msa", "frank-wolfe", "cfw", "bfw"]:
             self.assignment = LinearApproximation(self, algorithm.lower())
         else:
             raise Exception('Algorithm not listed in the case selection')
 
+        self.__dict__['algorithm'] = algorithm
```

Comment:
I think this is why your test is failing, maybe set algorithm to 'frank-wolfe' if the user provided fw?

Reply:
On the other hand special treatment is a bit annoying (as proven by the failing test :) ). Maybe we should wrap this up into its own class?

Reply:
Not too important right now, but if we ever add other assignment methods we might have to

Reply:
special treatment is fine... as long as you just do it once.
do what Jan said, if algo == "fw" then replace algo = "frank-wolfe"

After that point never allow fw, only frank-wolfe

Reply:
I think it is solved appropriately now.